src/components/DatePicker.js

```diff
@@ -11,24 +11,81 @@ class DatePicker extends React.Component {
     super(props);
     this.state = {
       currentDate: null,
-      selectedDate: moment.unix(this.props.defaultDate),
+      inputValue: this._getInputValueByDate(this.props.defaultDate),
+      isValid: true,
+      selectedDate: this.props.defaultDate,
       showCalendar: false
     };
   }
 
+  _getInputValueByDate (date) {
+    let inputValue = null;
+
+    if (date) {
+      const newDate = moment.unix(date);
+
+      if (newDate.isValid()) {
+        inputValue = newDate.format(this.props.format);
+      } else {
+        inputValue = date;
+      }
+    }
+
+    return inputValue;
+  }
+
+  _getSelectedDate () {
+    const selectedDate = this.state.selectedDate;
+
+    return selectedDate && moment.unix(selectedDate).isValid() ? this.state.selectedDate : moment().unix();
+  }
+
   _handleDateSelect (date) {
     if (this.props.closeOnDateSelect) {
-      this._handleBlur();
+      this._handleScrimClick();
     }
+
     this.setState({
-      selectedDate: moment.unix(date)
+      inputValue: moment.unix(date).format(this.props.format),
+      isValid: true,
+      selectedDate: date
     });
 
     this.props.onDateSelect(date);
   }
 
+  _handleInputBlur (evt) {
+    let inputValue = null;
+    let isValid = true;
+    let selectedDate = null;
+
+    if (evt.target.value && evt.target.value.length) {
+      const newDate = moment(new Date(evt.target.value));
+
+      inputValue = this._getInputValueByDate(newDate.isValid() ? newDate.unix() : evt.target.value);
+      isValid = newDate.isValid();
+      selectedDate = newDate.isValid() ? newDate.unix() : this.state.selectedDate;
+    }
+
+    this.setState({
+      inputValue,
+      isValid,
+      selectedDate
+    });
+
+    if (isValid) {
+      this.props.onDateSelect(selectedDate);
+    }
+  }
+
+  _handleInputChange (evt) {
+    this.setState({
+      inputValue: evt.target.value
+    });
+  }
+
   _handlePreviousClick () {
-    const selectedDate = moment(this.state.selectedDate).locale(this.props.locale);
+    const selectedDate = moment.unix(this._getSelectedDate()).locale(this.props.locale);
```

Comment:
No, `_getSelectedDate` is returning a unix time stamp.  See line 40.  It either returns the existing unix time stamp contained in this.state.selectedDate or a new moment for today as a unix time stamp.